### PR TITLE
(fix) Authoring: Minor fixes

### PR DESCRIPTION
**Pull request by mdhaman**
- Fixing the not for publication flag.
- Strip html from the headline and abstract
- Fixing the unit test.

### CODE REVIEW

File: client/app/scripts/superdesk-archive/views/media-box-list.html [hunk 1]
```diff
@@ -32,7 +32,7 @@
             <div class="takekey" ng-if="item.anpa_take_key">{{ item.anpa_take_key }}</div>
             <div class="category" ng-if="item['anpa-category'].name">{{ item['anpa-category'].name }}</div>
             <div class="provider" sd-meta-ingest data-item="item"></div>
-            <div ng-if="item.marked_for_not_publication == true" class="not_for_publication" translate>Not for publication</div>
+            <div ng-if="item.marked_for_not_publication" class="state-label not-for-publication">Not For Publication</div>
```
Comment (plamut) [inline client/app/scripts/superdesk-archive/views/media-box-list.html]: Was this `translate` here deleted by an accident?

File: client/app/scripts/superdesk-archive/views/metadata-view.html [hunk 2]
```diff
@@ -1,7 +1,10 @@
 <div class="metadata-view">
     <dl>
         <dt translate>State</dt>
-        <dd><div class="state-label state-{{item.state}}">{{item.state}}</div></dd>
+        <dd>
+            <div class="state-label state-{{item.state}}">{{item.state}}</div>
+            <div ng-if="item.marked_for_not_publication" class="state-label not-for-publication">Not For Publication</div>
```
Comment (plamut) [inline client/app/scripts/superdesk-archive/views/metadata-view.html]: Missing `translate`?

**Comment by plamut** [comment]
I think it's good (apart from a few missing  `translate` directives).
:+1:

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 3]
```diff
@@ -722,6 +728,15 @@
                     }
                 }, true);
 
+                $scope.$watch('item.marked_for_not_publication', function(newValue, oldValue) {
+                    if (newValue !== oldValue) {
```
Comment (petrjasek) [inline client/app/scripts/superdesk-authoring/authoring.js]: imo this check is extra, if this is not true, watch shouldn't be triggered

**Comment by petrjasek** [comment]
:+1:

File: client/app/scripts/superdesk-authoring/authoring.js (hunk 3 above)
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: Actually a watch _can_ be triggered in the initialization phase with both old and new values equal. If triggering the handler is not desired in such case, this type of check is necessary.

**Comment by amagdas** [comment]
:+1:

**Pull request merged**